src/main/java/com/bouncestorage/glacierproxy/GlacierProxy.java

```diff
@@ -50,25 +52,61 @@ public Job getJobHandler(Map<String, String> parameters) {
         return new Job(this);
     }
 
+    public Multipart getMultipartHandler(Map<String, String> parameters) {
+        return new Multipart(this);
+    }
+
     public JSONObject getJob(String vault, UUID jobId) {
-        if (!jobMap.containsKey(vault)) {
-            return null;
-        }
-        if (!jobMap.get(vault).containsKey(jobId)) {
+        try {
+            return jobMap.get(vault).get(jobId);
+        } catch (NullPointerException npe) {
```

Comment:
You should not want to catch `NullPointerException` since many things could cause it, e.g., null `jobMap`.  Instead consider the following to avoid NPE and TOCTOU:

```java
Map<UUID, JSONObject> map = jobMap.get(vault);
if (map == null) {
    return null;
}
return map.get(jobId);
```